### PR TITLE
Add join table for visits

### PR DESCRIPTION
**Pull request by KareemWilson**
In this Branch I did the following: 

1. Create vets, specializations, visits tables
2. Write queries to answer the following:

- Who was the last animal seen by William Tatcher?
Who was the last animal seen by William Tatcher?

- How many different animals did Stephanie Mendez see?
How many different animals did Stephanie Mendez see?

- List all vets and their specialties, including vets with no specialties.
List all vets and their specialties, including vets with no specialties

- List all animals that visited Stephanie Mendez between April 1st and August 30th, 2020.
List all animals that visited Stephanie Mendez between April 1st and August 30th, 2020

- What animal has the most visits to vets?
What animal has the most visits to vets?

- Who was Maisy Smith's first visit?
Who was Maisy Smith's first visit?

- Details for most recent visit: animal information, vet information, and date of visit.
Details for most recent visit: animal information, vet information, and date of visit

- How many visits were with a vet that did not specialize in that animal's species?
How many visits were with a vet that did not specialize in that animal's species?

- What specialty should Maisy Smith consider getting? Look for the species she gets the most.
What specialty should Maisy Smith consider getting? Look for the species she gets the most

### CODE REVIEW

File: queries.sql [hunk 1]
```diff
@@ -125,3 +125,95 @@ SELECT o.full_name, count(a.name) as number_of_animals
...
+-- Who was the last animal seen by William Tatcher?
+SELECT a.name AS "Name of Animal", vets.name AS "Vet", date_of_visit FROM visits v LEFT JOIN vets ON
+v.vet_id = vets.id LEFT JOIN animals a ON v.animal_id = a.id WHERE vets.name = 'William Tatcher' ORDER BY date_of_visit DESC LIMIT 1;
+
+-- How many different animals did Stephanie Mendez see?
+SELECT vets.name AS "Vet",COUNT(*) FROM visits v LEFT JOIN vets ON v.vet_id = vets.id 
+    LEFT JOIN animals a
+    ON v.animal_id = a.id 
+    WHERE vets.name = 'Stephanie Mendez' 
+    GROUP BY vets.name;
```
Comment (danmainah) [inline queries.sql:141]: - [x] @great work on your query however your query result is wrong Stephanie Mendez visited 4 animals as per the data provided,
image

File: queries.sql [hunk 2]
```diff
@@ -125,3 +125,95 @@ SELECT o.full_name, count(a.name) as number_of_animals
...
+SELECT a.name AS "Animals", vets.name AS "Vet" 
+    FROM visits v 
+    LEFT JOIN animals a 
+    ON v.animal_id = a.id
+    LEFT JOIN vets 
+    ON v.vet_id = vets.id 
+    WHERE vets.name = 'Stephanie Mendez' 
+    AND date_of_visit 
+    BETWEEN '2020-04-01' AND '2020-08-30' 
+    GROUP BY a.name, vets.name;
```
Comment (danmainah) [inline queries.sql]: - [x]  @great work on your query ,however, your query result is wrong Stephanie Mendez visited 2 animals as per the data provided,
- Agumon visited Stephanie Mendez on Jul 22th, 2020.
- Blossom visited Stephanie Mendez on May 24th, 2020.

File: queries.sql [hunk 3]
```diff
@@ -125,3 +125,95 @@ SELECT o.full_name, count(a.name) as number_of_animals
...
+-- Details for most recent visit: animal information, vet information, and date of visit.
+SELECT vets.name AS "Vet",a.name AS "Animal", date_of_visit 
+    FROM visits v 
+    LEFT JOIN animals a 
+    ON v.animal_id = a.id 
+    LEFT JOIN vets 
+    ON v.vet_id = vets.id 
+    WHERE vets.name = 'Maisy Smith' 
+    ORDER BY date_of_visit 
+    LIMIT 1;
```
Comment (danmainah) [inline queries.sql]: - [x] @great work on your query,however, your query result is wrong the recent visit was by: -
Angemon visited Jack Harkness on Nov 4th, 2020.

**Review by danmainah — CHANGES_REQUESTED** [review]
Hi @KareemWilson , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✔️ Gitflow was correctly used
✔️  Good design


# Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Comment by KareemWilson** [comment]
I adjust data that I inserted before, because it was Wrong, 
**NOW it works correctly :+1:** 

- How many different animals did Stephanie Mendez see?
How many different animals did Stephanie Mendez see?

- List all animals that visited Stephanie Mendez between April 1st and August 30th, 2020.
List all animals that visited Stephanie Mendez between April 1st and August 30th, 2020

- Details for most recent visit: animal information, vet information, and date of visit.
Details for most recent visit: animal information, vet information, and date of visit

**Review by BertrandConxy — APPROVED** [review]
Hi @KareemWilson, 

In my opinion, your project is ready for the presentation!  There is nothing else to say other than... it's time to merge it :shipit: 

## Highlights 💪 
- no linter errors
- project requirements are met
- good PR description
- good README


Good luck! 🎉 


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**Pull request merged**